commit 0047ca0a45c6a481abd467fb52d2a480ffc8c6b9
Author: Paul Ready <lxtwin@homecall.co.uk>
Date:   Wed Oct 29 14:25:50 2008 -0700

    USB: add Nikon D300 camera to unusual_devs
    
    Addresses http://bugzilla.kernel.org/show_bug.cgi?id=11685
    
    When A Nikon D300 camera is connected to a system it is seen in
    /proc/bus/pci/devices but is not accessible.
    
    This is seen in the above file:
    
    T:  Bus=01 Lev=01 Prnt=01 Port=05 Cnt=03 Dev#= 11 Spd=480 MxCh= 0
    D:  Ver= 2.00 Cls=00(>ifc ) Sub=00 Prot=00 MxPS=64 #Cfgs=  1
    P:  Vendor=04b0 ProdID=041a Rev= 1.03
    S:  Manufacturer=NIKON
    S:  Product=NIKON DSC D300
    S:  SerialNumber=000008014379
    C:* #Ifs= 1 Cfg#= 1 Atr=c0 MxPwr=  2mA
    I:* If#= 0 Alt= 0 #EPs= 3 Cls=06(still) Sub=01 Prot=01 Driver=usbfs
    E:  Ad=81(I) Atr=02(Bulk) MxPS= 512 Ivl=0ms
    E:  Ad=02(O) Atr=02(Bulk) MxPS= 512 Ivl=0ms
    E:  Ad=83(I) Atr=03(Int.) MxPS=   8 Ivl=32ms
    
    Cc: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit 48e1a540e1e9ea62a2b3088a69ecf934f3172b14
Author: Ozan Sener <themgzzy@gmail.com>
Date:   Mon Dec 8 19:15:45 2008 +0200

    USB: Unusual devs patch for Nokia 3500c
    
    T:  Bus=02 Lev=01 Prnt=01 Port=00 Cnt=01 Dev#=  3 Spd=12  MxCh= 0
    D:  Ver= 2.00 Cls=00(>ifc ) Sub=00 Prot=00 MxPS=64 #Cfgs=  1
    P:  Vendor=0421 ProdID=0060 Rev= 5.51
    S:  Manufacturer=Nokia
    S:  Product=Nokia 3500c
    S:  SerialNumber=357687010280751
    C:* #Ifs= 1 Cfg#= 1 Atr=c0 MxPwr=100mA
    I:* If#= 0 Alt= 0 #EPs= 2 Cls=08(stor.) Sub=06 Prot=50 Driver=usb-storage
    E:  Ad=81(I) Atr=02(Bulk) MxPS=  64 Ivl=0ms
    E:  Ad=01(O) Atr=02(Bulk) MxPS=  64 Ivl=0ms
    
    From: Ozan Sener <themgzzy@gmail.com>
    Cc: stable <stable@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit ed4103b3fcf38985995e732dab6c3e2b9693f6cb
Author: Ricky Wong <evilbladewarrior@gmail.com>
Date:   Tue Nov 4 19:13:45 2008 +0800

    usb: unusual devs patch for Nokia 7610 Supernova
    
    Additional sectors were reported by the Nokia 7610 Supernova phone in
    usb storage mode. The following patch rectifies the aforementioned
    problem.
    
    Signed-off-by: Ricky Wong Yung Fei <evilbladewarrior@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit a6b7b034d7f20761c55743be2acb762ce09a0c6b
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Fri Nov 21 16:15:12 2008 -0500

    USB: storage: unusual_devs entry for Mio C520-GPS
    
    This patch (as1176) adds an unusual_devs entry for the Mio C520 GPS
    unit.  Other devices also based on the Mitac hardware use the same USB
    interface firmware, so the Vendor and Product names are generalized.
    
    This fixes Bugzilla #11583.
    
    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Tested-by: Tamas Kerecsen <kerecsen@bigfoot.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit 621b239d75b790ac66854d46b094874f69e6776e
Author: Tobias Kunze Briseño <t@fictive.com>
Date:   Mon Nov 24 11:28:31 2008 -0500

    USB: usb-storage: unusual_devs entry for Nikon D2H
    
    This patch adds an unusual_devs entry for the Nikon D2H camera.
    
    From: Tobias Kunze Briseño <t@fictive.com>,
    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit e8fab4ce763c36869624c5388714ff19c30a91a7
Author: Frédéric Marchal <frederic.marchal@wowcompany.com>
Date:   Mon Oct 13 14:00:11 2008 +0200

    USB: Unusual dev for Mio moov 330 gps
    
    Here is an entry for the unusual_devs.h file to handle a Mio Moov 330 GPS that
    stops responding when it is requested to transfer more than 64KB. The patch is
    taken against kernel-2.6.27-git3.
    
    Signed-off-by: Frédéric Marchal <frederic.marchal@wowcompany.co
    Signed-off-by: Phil Dibowitz <phil@ipom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit af747c460b663a4b7795dc3f6897b5506fde6888
Author: David Almaroad <dalmaroad@gmail.com>
Date:   Mon Sep 22 15:00:11 2008 -0700

    usb: unusual devs patch for Nokia 5310 Music Xpress
    
    The Nokia 5310 Music Xpress phone reports one too many sectors in
    usb-storage mode.  This patch resolves that.
    
    Signed-off-by: David Almaroad <dalmaroad@gmail.com>
    Cc: Matthew Dharm <mdharm-usb@one-eyed-alien.net>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit d28525f8866244a77b677c0470d4859cb2b6c2a8
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Thu Aug 7 13:02:40 2008 -0400

    usb-storage: unusual_devs entry for Nokia 5300
    
    This patch (as1120) adds an unusual_devs entry for the Nokia 5300.
    Maybe once Nokia releases the Symbian code we'll be able to fix all
    the problems it has with the USB mass-storage protocol.
    
    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Tested-by: Cedric Godin <cedric@belbone.be>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit 589afd3bec907f02c133d7b8185b8af534f14a8e
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Mon Nov 17 14:32:16 2008 -0500

    USB: storage: update unusual_devs entries for Nokia 5300 and 5310
    
    This patch (as1168) updates the unusual_devs entry for the Nokia 5300.
    According to Jorge Lucangeli Obes <t4m5yn@gmail.com>, some existing
    models have a revision number lower than the lower limit of the
    current entry.
    
    The patch also moves the entry for the Nokia 5310 to its correct place
    in the file.
    
    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit 1393fce7189427bdb4d6115ca5566ca8d0fc86f3
Author: CSÉCSY László <boobaa@frugalware.org>
Date:   Tue Dec 9 23:39:14 2008 +0100

    USB: storage: unusual_devs.h: Nokia 3109c addition
    
    2.6.26(.x, cannot remember) could handle the microSD card in my Nokia
    3109c attached via USB as mass storage, 2.6.27(.x, up to and included
    2.6.27.8) cannot. Please find the attached patch which fixes this
    regression, and a copy of /proc/bus/usb/devices with my phone plugged in
    running with this patch on Frugalware.
    
    T:  Bus=02 Lev=01 Prnt=01 Port=01 Cnt=02 Dev#=  4 Spd=12  MxCh= 0
    D:  Ver= 2.00 Cls=00(>ifc ) Sub=00 Prot=00 MxPS=64 #Cfgs=  1
    P:  Vendor=0421 ProdID=0063 Rev= 6.01
    S:  Manufacturer=Nokia
    S:  Product=Nokia 3109c
    S:  SerialNumber=359561013742570
    C:* #Ifs= 1 Cfg#= 1 Atr=c0 MxPwr=100mA
    I:* If#= 0 Alt= 0 #EPs= 2 Cls=08(stor.) Sub=06 Prot=50 Driver=usb-storage
    E:  Ad=81(I) Atr=02(Bulk) MxPS=  64 Ivl=0ms
    E:  Ad=01(O) Atr=02(Bulk) MxPS=  64 Ivl=0ms
    
    From: CSÉCSY László <boobaa@frugalware.org>
    Cc: Phil Dibowitz <phil@ipom.com>
    Cc: stable <stable@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit a4b188095912eee83d065f000dfe06f25919750b
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Mon Dec 1 10:23:43 2008 -0500

    usb-storage: update unusual_devs entry for Nokia 5310
    
    This patch (as1179) updates the unusual_devs entry for Nokia's 5310
    phone to include a more recent firmware revision.
    
    This fixes Bugzilla #12099.
    
    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Tested-by: Robson Roberto Souza Peixoto <robsonpeixoto@gmail.com>
    Cc: stable <stable@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

commit f9dc8f99e5846606b1f475b3905eaf5ae1017c50
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Tue Nov 18 14:08:07 2008 -0500

    usb-storage: clean up unusual_devs.h
    
    This patch (as1170) removes some duplicate entries in unusual_devs.h
    and rearranges a few others to put the list in proper numerical order.
    
    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

Adjusted to apply to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN a/drivers/usb/storage/unusual_devs.h b/drivers/usb/storage/unusual_devs.h
--- a/drivers/usb/storage/unusual_devs.h	2009-03-20 16:54:45.000000000 -0600
+++ b/drivers/usb/storage/unusual_devs.h	2009-03-20 16:55:37.000000000 -0600
@@ -162,6 +162,28 @@ UNUSUAL_DEV(  0x0421, 0x0019, 0x0592, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_MAX_SECTORS_64 ),
 
+/* Reported by CSECSY Laszlo <boobaa@frugalware.org> */
+UNUSUAL_DEV(  0x0421, 0x0063, 0x0001, 0x0601,
+		"Nokia",
+		"Nokia 3109c",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY ),
+
+/* Patch for Nokia 5310 capacity */
+UNUSUAL_DEV(  0x0421, 0x006a, 0x0000, 0x0701,
+		"Nokia",
+		"5310",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY ),
+
+/* Submitted by Ricky Wong Yung Fei <evilbladewarrior@gmail.com> */
+/* Nokia 7610 Supernova - Too many sectors reported in usb storage mode */
+UNUSUAL_DEV(  0x0421, 0x00f5, 0x0000, 0x0470,
+		"Nokia",
+		"7610 Supernova",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY ),
+
 /* Reported by Mario Rettig <mariorettig@web.de> */
 UNUSUAL_DEV(  0x0421, 0x042e, 0x0100, 0x0100,
 		"Nokia",
@@ -227,6 +249,13 @@ UNUSUAL_DEV(  0x0421, 0x0495, 0x0370, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_MAX_SECTORS_64 ),
 
+/* Reported by Cedric Godin <cedric@belbone.be> */
+UNUSUAL_DEV(  0x0421, 0x04b9, 0x0500, 0x0551,
+		"Nokia",
+		"5300",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY ),
+
 /* Reported by Olaf Hering <olh@suse.de> from novell bug #105878 */
 UNUSUAL_DEV(  0x0424, 0x0fdc, 0x0210, 0x0210,
 		"SMSC",
@@ -277,6 +306,18 @@ UNUSUAL_DEV(  0x045a, 0x5210, 0x0101, 0x
 		US_SC_SCSI, US_PR_KARMA, rio_karma_init, 0),
 #endif
 
+/* Reported by Tamas Kerecsen <kerecsen@bigfoot.com>
+ * Obviously the PROM has not been customized by the VAR;
+ * the Vendor and Product string descriptors are:
+ *	Generic Mass Storage (PROTOTYPE--Remember to change idVendor)
+ *	Generic Manufacturer (PROTOTYPE--Remember to change idVendor)
+ */
+UNUSUAL_DEV(  0x045e, 0xffff, 0x0000, 0x0000,
+		"Mitac",
+		"GPS",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_MAX_SECTORS_64 ),
+
 /*
  * This virtual floppy is found in Sun equipment (x4600, x4200m2, etc.)
  * Reported by Pete Zaitcev <zaitcev@redhat.com>
@@ -321,6 +362,15 @@ UNUSUAL_DEV(  0x04a4, 0x0004, 0x0001, 0x
 		"DVD-CAM DZ-MV100A Camcorder",
 		US_SC_SCSI, US_PR_CB, NULL, US_FL_SINGLE_LUN),
 
+/* BENQ DC5330
+ * Reported by Manuel Fombuena <mfombuena@ya.com> and
+ * Frank Copeland <fjc@thingy.apana.org.au> */
+UNUSUAL_DEV(  0x04a5, 0x3010, 0x0100, 0x0100,
+		"Tekom Technologies, Inc",
+		"300_CAMERA",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_IGNORE_RESIDUE ),
+
 /* Patch for Nikon coolpix 2000
  * Submitted by Fabien Cosse <fabien.cosse@wanadoo.fr>*/
 UNUSUAL_DEV(  0x04b0, 0x0301, 0x0010, 0x0010,
@@ -336,6 +386,13 @@ UNUSUAL_DEV(  0x04b0, 0x0401, 0x0200, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_FIX_CAPACITY),
 
+/* Reported by Tobias Kunze Briseno <t-linux@fictive.com> */
+UNUSUAL_DEV(  0x04b0, 0x0403, 0x0200, 0x0200,
+		"NIKON",
+		"NIKON DSC D2H",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY),
+
 /* Reported by Milinevsky Dmitry <niam.niam@gmail.com> */
 UNUSUAL_DEV(  0x04b0, 0x0409, 0x0100, 0x0100,
 		"NIKON",
@@ -392,6 +449,13 @@ UNUSUAL_DEV(  0x04b0, 0x0417, 0x0100, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_FIX_CAPACITY),
 
+/* Reported by paul ready <lxtwin@homecall.co.uk> */
+UNUSUAL_DEV(  0x04b0, 0x0419, 0x0100, 0x0200,
+		"NIKON",
+		"NIKON DSC D300",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY),
+
 /* Reported by Doug Maxey (dwm@austin.ibm.com) */
 UNUSUAL_DEV(  0x04b3, 0x4001, 0x0110, 0x0110,
 		"IBM",
@@ -399,15 +463,6 @@ UNUSUAL_DEV(  0x04b3, 0x4001, 0x0110, 0x
 		US_SC_DEVICE, US_PR_CB, NULL,
 		US_FL_MAX_SECTORS_MIN),
 
-/* BENQ DC5330
- * Reported by Manuel Fombuena <mfombuena@ya.com> and
- * Frank Copeland <fjc@thingy.apana.org.au> */
-UNUSUAL_DEV(  0x04a5, 0x3010, 0x0100, 0x0100,
-		"Tekom Technologies, Inc",
-		"300_CAMERA",
-		US_SC_DEVICE, US_PR_DEVICE, NULL,
-		US_FL_IGNORE_RESIDUE ),
-
 #ifdef CONFIG_USB_STORAGE_CYPRESS_ATACB
 /* CY7C68300 : support atacb */
 UNUSUAL_DEV(  0x04b4, 0x6830, 0x0000, 0x9999,
@@ -726,15 +781,15 @@ UNUSUAL_DEV(  0x054c, 0x006d, 0x0000, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_FIX_INQUIRY ),
 
-/* Submitted by Mike Alborn <malborn@deandra.homeip.net> */
-UNUSUAL_DEV(  0x054c, 0x016a, 0x0000, 0x9999,
+/* Submitted by Frank Engel <frankie@cse.unsw.edu.au> */
+UNUSUAL_DEV(  0x054c, 0x0099, 0x0000, 0x9999,
 		"Sony",
 		"PEG Mass Storage",
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_FIX_INQUIRY ),
-		
-/* Submitted by Frank Engel <frankie@cse.unsw.edu.au> */
-UNUSUAL_DEV(  0x054c, 0x0099, 0x0000, 0x9999,
+
+/* Submitted by Mike Alborn <malborn@deandra.homeip.net> */
+UNUSUAL_DEV(  0x054c, 0x016a, 0x0000, 0x9999,
 		"Sony",
 		"PEG Mass Storage",
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
@@ -1373,6 +1428,16 @@ UNUSUAL_DEV( 0x0c0b, 0xa109, 0x0000, 0xf
 		US_FL_SINGLE_LUN ),
 #endif
 
+/*
+ * Pete Zaitcev <zaitcev@yahoo.com>, bz#164688.
+ * The device blatantly ignores LUN and returns 1 in GetMaxLUN.
+ */
+UNUSUAL_DEV( 0x0c45, 0x1060, 0x0100, 0x0100,
+		"Unknown",
+		"Unknown",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_SINGLE_LUN ),
+
 /* Submitted by: Nick Sillik <n.sillik@temple.edu>
  * Needed for OneTouch extension to usb-storage
  *
@@ -1390,16 +1455,6 @@ UNUSUAL_DEV( 0x0c0b, 0xa109, 0x0000, 0xf
 			0),
 #endif
 
-/*
- * Pete Zaitcev <zaitcev@yahoo.com>, bz#164688.
- * The device blatantly ignores LUN and returns 1 in GetMaxLUN.
- */
-UNUSUAL_DEV( 0x0c45, 0x1060, 0x0100, 0x0100,
-		"Unknown",
-		"Unknown",
-		US_SC_DEVICE, US_PR_DEVICE, NULL,
-		US_FL_SINGLE_LUN ),
-
 /* Submitted by Joris Struyve <joris@struyve.be> */
 UNUSUAL_DEV( 0x0d96, 0x410a, 0x0001, 0xffff,
 		"Medion",
@@ -1539,13 +1594,6 @@ UNUSUAL_DEV(  0x0fce, 0xe030, 0x0000, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_FIX_CAPACITY | US_FL_IGNORE_RESIDUE ),
 
-/* Reported by Ricardo Barberis <ricardo@dattatec.com> */
-UNUSUAL_DEV(  0x0fce, 0xe092, 0x0000, 0x0000,
-		"Sony Ericsson",
-		"P1i",
-		US_SC_DEVICE, US_PR_DEVICE, NULL,
-		US_FL_IGNORE_RESIDUE ),
-
 /* Reported by Emmanuel Vasilakis <evas@forthnet.gr> */
 UNUSUAL_DEV(  0x0fce, 0xe031, 0x0000, 0x0000,
 		"Sony Ericsson",
@@ -1553,6 +1601,13 @@ UNUSUAL_DEV(  0x0fce, 0xe031, 0x0000, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_IGNORE_RESIDUE | US_FL_FIX_CAPACITY ),
 
+/* Reported by Ricardo Barberis <ricardo@dattatec.com> */
+UNUSUAL_DEV(  0x0fce, 0xe092, 0x0000, 0x0000,
+		"Sony Ericsson",
+		"P1i",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_IGNORE_RESIDUE ),
+
 /* Reported by Kevin Cernekee <kpc-usbdev@gelato.uiuc.edu>
  * Tested on hardware version 1.10.
  * Entry is needed only for the initializer function override.
@@ -1779,6 +1834,15 @@ UNUSUAL_DEV(  0x2770, 0x915d, 0x0010, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_FIX_CAPACITY ),
 
+/* Reported by Frederic Marchal <frederic.marchal@wowcompany.com>
+ * Mio Moov 330
+ */
+UNUSUAL_DEV(  0x3340, 0xffff, 0x0000, 0x0000,
+		"Mitac",
+		"Mio DigiWalker USB Sync",
+		US_SC_DEVICE,US_PR_DEVICE,NULL,
+		US_FL_MAX_SECTORS_64 ),
+
 /* Reported by Andrey Rahmatullin <wrar@altlinux.org> */
 UNUSUAL_DEV(  0x4102, 0x1020, 0x0100,  0x0100,
 		"iRiver",
